com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/util/ProblemsViewTestUtil.java

```diff
@@ -0,0 +1,166 @@
+    bot.viewById(PROBLEMS_VIEW).show();
+    bot.viewById(PROBLEMS_VIEW).setFocus();
+  }
+
+  /**
+   * Gets the markers tree.
+   *
+   * @param bot
+   *          the bot, must not be {@code null}
+   * @return the markers tree, never {@code null}
+   */
+  public static SWTBotTree getMarkersTree(final SwtWorkbenchBot bot) {
+    showProblemsView(bot);
+    return bot.viewById(PROBLEMS_VIEW).bot().tree(0);
+  }
+
+  /**
+   * Convenience method which clicks on the menu item found. See {@link DynamicViewMenu.click}.
+   *
+   * @param bot
+   *          the bot, must not be {@code null}
+   * @param menuPath
+   *          the menu path, must not be {@code null} nor empty
+   */
+  public static void clickMenuItem(final SwtWorkbenchBot bot, final String... menuPath) {
+    showProblemsView(bot);
+    DynamicViewMenu.create(bot).click(menuPath);
+  }
+
+  /**
+   * Groups by none in Problems view.
+   *
+   * @param bot
+   *          the bot, must not be {@code null}
+   */
+  public static void groupByNone(final SwtWorkbenchBot bot) {
+    clickMenuItem(bot, GROUP_BY, NONE);
+  }
+
+  /**
+   * Show all errors in Problems view.
+   *
+   * @param bot
+   *          the bot, must not be {@code null}
+   */
+  public static void showAllErrors(final SwtWorkbenchBot bot) {
+    showProblemsView(bot);
+    // Select "Show All" first, to avoid toggling "All Errors" off
+    showAll(bot);
+    clickMenuItem(bot, SHOW, ALL_ERRORS);
+  }
+
+  /**
+   * Show all markers in Problems view.
+   *
+   * @param bot
+   *          the bot, must not be {@code null}
+   */
+  public static void showAll(final SwtWorkbenchBot bot) {
+    showProblemsView(bot);
+    clickMenuItem(bot, SHOW, SHOW_ALL);
+  }
+
+  /**
+   * Bulk-apply the common quickfix to multiple markers.
+   * Autobuilding should be disabled, to avoid losing focus while selecting markers.
+   *
+   * @param bot
+   *          the bot, must not be {@code null}
+   * @param quickfixLabel
+   *          label of the quickfix to apply, must not be {@code null}
+   * @param markers
+   *          markers to fix, must not be {@code null}
+   */
+  @SuppressWarnings("unchecked")
+  public static void bulkApplyQuickfix(final SwtWorkbenchBot bot, final String quickfixLabel, final SWTBotTreeItem... markers) {
+
+    // Select markers. Keep retrying if focus is lost at an inopportune moment.
+    final SWTBotTree markersTreeBot = getMarkersTree(bot);
+    do {
+      markersTreeBot.select(markers);
+    } while (markersTreeBot.selectionCount() != markers.length);
+
+    // Open the Quick Fix dialog
+    ContextActionUiTestUtil.clickContextMenu(markersTreeBot, QUICK_FIX_CONTEXT_MENU_ITEM_LABEL);
+    bot.waitUntil(Conditions.shellIsActive(QUICK_FIX_DIALOG_TEXT));
+
+    // Select the quickfix
+    bot.waitUntil(Conditions.waitForWidget(WidgetMatcherFactory.allOf(WidgetMatcherFactory.widgetOfType(Table.class), WidgetMatcherFactory.withLabel(SELECT_A_FIX_TABLE_LABEL))));

```

Comment:
These look redundant since the next line runs a similar wait condition before returning a SWTBot table widget.

Reply:
Redundant bot.waitUntil calls removed.
I've kept the wait for shellIsActive, so that if the dialog fails to open then the test will fail meaningfully.